src/main/kotlin/test/shapes/TestTriangle.kt

```diff
@@ -0,0 +1,82 @@
+package test.shapes
+
+import spr5.scene.Coordinate
+import spr5.scene.SceneTriangle
+import spr5.scene.Rgba
+import test.TestClass
+import spr5.util.*
+
+
+class TestTriangle() : TestClass() {
+    private var v1: Array<Double> = emptyArray();
+    private var triangleColor : Rgba = setColor();
+    private var testTriangle : SceneTriangle = setUp();
+
+
+    @Before
+    fun setUp() : SceneTriangle {
+        val point1 = Coordinate(50.0f, 50.0f, 50.0f)
+        val point2 = Coordinate (100.0f, 100.0f, 100.0f)
+        val point3 = Coordinate (-50.0f, -50.0f, -50.0f)
+        val vertices = arrayOf(
+                point1, point2, point3)
+        return SceneTriangle(vertices, this.triangleColor);
+    }
+
+    @After
+    fun tearDown() {
+    }
+    fun setColor() : Rgba {
+        return Rgba(1.0f, 0.5f, 0.7f, 0.5f);
+    }
+
+    public fun TestObjectCreation(){
+        assertNotNull(setUp())
+    }
+
+
+    public fun TestTriangleColor(){
+        val wrongTestColor = Rgba(1.0f, 0.5f, 0.7f, 1.0f);
+        val wrongObjectColor = Rgba(1.0f, 0.5f, 0.7f, 0.5f);
+        assertNotEquals(wrongTestColor, testTriangle.color)
+        assertNotEquals(wrongObjectColor, testTriangle.color)
+        assertEquals(triangleColor, testTriangle.color)
```

Comment:
The effective assertion call is: 
```
triangleColor == testTriangle.color
```

I'm not sure if real structural equality is compared in this case because triangleColor and testTriangle.color are in this code the same reference. We would need to override Rgba::equals (https://kotlinlang.org/docs/reference/equality.html)